doc/espnet2_tutorial.md

```diff
@@ -346,3 +346,243 @@ To enable online decoding, the argument `--use_streaming true` should be added t
+    search_type: maes
+    nstep: Number of maximum expansion steps at each time step (int, default = 2)
+    expansion_gamma: Number of additional candidates in expanded hypotheses selection. (int, default = 2)
+    expansion_beta: Allowed logp difference for prune-by-value method. (float, default = 2.3)
+
+***Note:*** Except for the default algorithm, the described parameters are used to control the performance and decoding speed. The optimal values for each parameter are task-dependent; a high value will typically increase decoding time to focus on performance while a low value will improve decoding time at the expense of performance.
+
+***Note 2:*** The algorithms in the standalone version are the same as the one in the other version.. However, due to design choices, some parts were reworked and minor optimizations were added in the same time.
+
+### Streaming
+
+To enable streaming capabilities for Transducer models, we support dynamic chunk training and chunk-by-chunk decoding as proposed in [[Zhang et al., 2021]](https://arxiv.org/pdf/2012.05481.pdf). Our implementation is based on the version proposed in [Icefall](https://github.com/k2-fsa/icefall/), based itself on the original [WeNet](https://github.com/wenet-e2e/wenet/) one.
+
+For a complete explanation on the different procedure and parameters, we refer the reader to the corresponding paper.
+
+#### Training
+
+To train a streaming model, the parameter `dynamic_chunk_training` should be set to `True` in the encoder `main_conf`.
+
+From here, the user has access to two parameters in order to control the dynamic chunk selection (`short_chunk_threshold` and `short_chunk_size`) and another one to control the left context in the causal convolution and the attention module (`left_chunk_size`). All these parameters can be configured through the `main_conf`. The Encoder section provides a short description of the parameters.
+
+#### Decoding
+
+To perform chunk-by-chunk inference, the parameter `streaming` should be set to True in the decoding configuration(otherwise, offline decoding will be performed). Three parameters are available to control the decoding process:
+
+    chunk_size: Number of frames in chunk. (int, default = 16)
+    left_context: Number of frames in the left context. (int, default = 32)
+    right_context: Number of frames in the right context. (int, default = 0)
+
+For each parameter, the number of frames is defined AFTER subsampling, meaning the input chunk will be bigger than the one provided. The input size is determined by the frontend and the input block's subsampling, given `chunk_size + right_context` defining the decoding window.
+
+***Note:*** Because the training part does not consider the right context, relying on `right_context` during decoding may result in a mismatch and performance degration.
+***Note 2:*** All search algorithms but ALSD are available with chunk-by-chunk inference.
+
+### FAQ
+
+#### How to add a new block type to the custom encoder?
```

Comment:
@sw005320 I added a FAQ with a doc for new block type addition. Is this good enough?
Also, is there anything else I should add in the FAQ? @pengchengguo , @pyf98 ?

Reply:
It looks good to me.